elastic/logstash
# logging: move init into environment's settings post-processor

Ensures that the non-runner command line utilities like `bin/logstash-keystore`
correctly initialize the logger as-configured.

## Release notes

Fixed an issue where invoking the Logstash Keystore Utility (`bin/logstash-keystore`) incorrectly set up a logging directory to the literal `${sys:ls.logs}` and produced noise to the console about routing logs. This utility now correctly configures its logger using the provided settings file.

## What does this PR do?

Moves logger configuration "up" into the settings post-processor, which is invoked by both the normal pipeline runner _and_ the keystore utility `bin/logstash-keystore`. Now invoking either will correctly pick up logging configuration.

## Why is it important/What is the impact to the user?

Fixes an issue where invoking the Logstash Keystore Utility (`bin/logstash-keystore`) didn't route its logs correctly. Now the logs are where we expect them to be.

## Checklist




## How to test this PR locally

1. check out and assemble
2. Invoke `bin/logstash-keystore create`
3. Observe that the literal directory `${sys:ls.logs}` was not created in the current working directory
4. Optionally configure `path.logs`, `log.format` and/or `log.level` in your `config/logstash.yml`, and re-invoke the keystore utility. Observe that preferences are observed.
